### DISCUSSION TITLE
Is there any way to escape a variable template in a string?

### DISCUSSION THREAD

**Discussion by k33nice** [Orange-OpenSource/hurl#2425] [Q&A] [answered]
Say, I need to pass a string with `{{template}}` directly without hurl trying to replace it by variable `template`, e.g:
```
POST http://foo.test
{
  "template": "{\"foo\": \"some string with template {{some_var}}\"}"
}
```
I've tried `"{\"foo\": \"some string with template \{\{some_var\}\}"\"}`, but it not works.

The workaround I found is
```
POST http://foo.test
[Options]
variable: openBrace={
variable: closeBrace=}
{
  "template": "{\"foo\": \"some string with template {{openBrace}}{some_var}{{closeBrace}}\"}"
}
```
but it looks hacky.

**Comment by jcamiel** [comment, marked as answer]
Hi k33nice, it's not pretty for the moment, you can use unicode JSON literal :

- `{`=> \u007b
- `}`=> \u007d

So instead of 

```
POST https://foo.com
{ 
 "foo": "{{bar}}"
}
```

You can write:

```
POST https://foo.com
{ 
 "foo": "\u007b\u007bbar\u007d\u007d"
}
```

If you can test it on your side, I'm curious if it's working.


We've an issue to introduce a "raw" multiline string that won't do any variable resolution:

https://github.com/Orange-OpenSource/hurl/issues/1733

With this we should be able to write:

~~~
POST https://foo.com
Content-Type: application/json
```raw
{ 
 "foo": "{{bar}}"
}
```
~~~

We'll try to improve this!

**Reply by k33nice** [reply]
Appreciated! Yeah, unicode characters work as well. But personally, I found the approach with variables for braces more obvious (unless you have a unicode table in your mind :)).
The `raw` would be awesome for my case, but I guess someone would be needed to do both, e.g. ```"{\"tmpl\": \"\{\{escape\}\}\", \"var\": \"{{replace}}\"}"```. IDK, maybe it is hard to implement, these are just my thoughts on it.
Anyway, it's doable through variables or unicode characters so thanks again for the quick response!
